src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx

```diff
@@ -0,0 +1,163 @@
+import {
+  fetchPendingAppointments,
+  startNewAppointmentFlow,
+} from '../redux/actions';
+import { getRequestedAppointmentListInfo } from '../redux/selectors';
+import {
+  APPOINTMENT_STATUS,
+  FETCH_STATUS,
+  GA_PREFIX,
+} from '../../utils/constants';
+import { getVAAppointmentLocationId } from '../../services/appointment';
+import RequestListItem from './AppointmentsPageV2/RequestListItem';
+import NoAppointments from './NoAppointments';
+import InfoAlert from '../../components/InfoAlert';
+import { scrollAndFocus } from '../../utils/scrollAndFocus';
+import { selectFeatureStatusImprovement } from '../../redux/selectors';
+
+export default function RequestedAppointmentsListGroup({ hasTypeChanged }) {
+  const {
+    facilityData,
+    pendingAppointments,
+    pendingStatus,
+    showScheduleButton,
+  } = useSelector(
+    state => getRequestedAppointmentListInfo(state),
+    shallowEqual,
+  );
+  const featureStatusImprovement = useSelector(state =>
+    selectFeatureStatusImprovement(state),
+  );
+
+  const dispatch = useDispatch();
+
+  useEffect(
+    () => {
+      if (pendingStatus === FETCH_STATUS.notStarted) {
+        dispatch(fetchPendingAppointments());
+      } else if (hasTypeChanged && pendingStatus === FETCH_STATUS.succeeded) {
+        scrollAndFocus('#type-dropdown');
+      } else if (hasTypeChanged && pendingStatus === FETCH_STATUS.failed) {
+        scrollAndFocus('h3');
+      }
+    },
+    [pendingStatus, hasTypeChanged, dispatch],
+  );
+
+  if (
+    pendingStatus === FETCH_STATUS.loading ||
+    pendingStatus === FETCH_STATUS.notStarted
+  ) {
+    return (
+      <div className="vads-u-margin-y--8">
+        <va-loading-indicator
+          set-focus={hasTypeChanged}
+          message="Loading your appointment requests..."
+        />
+      </div>
+    );
+  }
+
+  if (pendingStatus === FETCH_STATUS.failed) {
+    return (
+      <InfoAlert
+        status="error"
+        headline="We’re sorry. We’ve run into a problem"
+      >
+        We’re having trouble getting your appointment requests. Please try again
+        later.
+      </InfoAlert>
+    );
+  }
+
+  // Group appointments by status
+  let appointmentsByStatus = pendingAppointments.reduce(
+    (previousValue, currentValue) => {
+      const groupByKey = currentValue.status;
+      const temp = previousValue;
+
+      // previousValue is initialize to an object
+      // Return the array of appointments if the status key exists or
+      // create a new status key with new array.
+      temp[groupByKey] = previousValue[groupByKey] || [];
+      temp[groupByKey].push(currentValue);
+      return temp;
+    },
+    {},
+  );
+
+  // Next, get an array of [key, value] pairs. Sort the array in decending order
+  // so that pending appointments are displayed 1st
+  appointmentsByStatus = Object.entries(appointmentsByStatus).sort((a, b) => {
+    if (a[0].toLowerCase() < b[0].toLocaleLowerCase()) return 1;
```

Comment:
@vbahinwillit - I am curious about toLowerCase and toLocaleLowerCase usage...what is the need to use the compares that way?

Reply:
Good catch. I only intended to use 'toLowerCase'.